### PR TITLE
Integration Test Script Pre-/Post-Test Cleanup

### PR DESCRIPTION
**Pull request by ihcsim**
Some improvement (hopefully) to the integration test script:

1. Remove existing resources prior to starting the test
1. Remove existing resources post upgrade test
1. Fail fast if 'install_test.go` fails
1. Don't perform cleanup if any of the tests fail to provide for debugging opportunity
1. Remove pre-test cleanup from `.travis.yaml`. This is now done in `bin/test-run` so that it can be shared between `staging` and `l5d-bot`.

Signed-off-by: Ivan Sim <ivan@buoyant.io>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -163,7 +164,6 @@ jobs:
         - |
           # Run integration tests.
           version="$(./linkerd version --client --short | tr -cd '[:alnum:]-')"
-          ./bin/test-cleanup
```
Comment (ihcsim) [inline .travis.yml]: This is now done in the `bin/test-run` script so that it can be shared between `staging` and `l5d-bot`.

File: bin/test-run [hunk 2]
```diff
@@ -59,15 +76,11 @@ function run_upgrade_test() {
     local stable_namespace="$1"
     local stable_version=$(curl -s https://versioncheck.linkerd.io/version.json | grep -o "stable-[0-9]*.[0-9]*.[0-9]*")
 
-    printf "Installing release [%s] namespace [%s]\n" "$stable_version" "$stable_namespace"
     install_stable $stable_namespace
-
-    printf "Upgrading release [%s] to [%s]\n" "$stable_version" "$linkerd_version"
-    run_test "$test_directory/install_test.go" --upgrade-from-version=$stable_version --linkerd-namespace=$stable_namespace
+    run_test "$test_directory/install_test.go" -failfast --upgrade-from-version=$stable_version --linkerd-namespace=$stable_namespace
```
Comment (ihcsim) [inline bin/test-run]: Fail fast for upgrade test. No point of continuing.

File: bin/test-run [hunk 3]
```diff
@@ -81,29 +94,34 @@ k8s_context=${3:-""}
...
+if [ $exit_code -ne 0 ]; then
+  printf "\\n=== FAIL: Can't upgrade to version $linkerd_version\\n"
+  exit $exit_code
 fi
+cleanup
 
-run_test "$test_directory/install_test.go" --linkerd-namespace=$linkerd_namespace || exit_code=$?
+# run the rest of the test suite. `cleanup` is not called if this test failed
+# so that there is a chance to debug the problem
+run_test "$test_directory/install_test.go" -failfast --linkerd-namespace=$linkerd_namespace || exit_code=$?
```
Comment (ihcsim) [inline bin/test-run]: Fail fast for `install_test.go`. No point of continuing.

File: .travis.yml [hunk 4]
```diff
@@ -7,6 +7,7 @@ sudo: false
 branches:
   only:
     - master
+    - isim/intg-test-script
```
Comment (ihcsim) [inline .travis.yml]: To be removed once testing is done.

Comment (alpeb) [reply .travis.yml]: Just a nit; `.travis.yml` will only be used when this branch gets merged, no?

Comment (alpeb) [reply .travis.yml]: Ivan clarified. This is for having Travis actually run this branch :+1:

**Pull request merged**